### PR TITLE
A new Typo checker repo action!

### PR DESCRIPTION
**Pull request by Chii-Onan**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature
- **Why this change was needed?** (You can also link to an open issue here)
After my previous typo fix PR, I thought that to prevent future typos in PR's  we should have this action as well!
- **Other information**:

### CODE REVIEW

**Comment by scopsy** [comment]
@Chii-Onan is it using cspell behind the scene? If we would love to add keywords to the vocabulary how would it be possible?

**Comment by scopsy** [comment]
Looks like there is quite a lot of errors :P Is there a way somehow to connect eslint with it aswell? So we can see the errors before they are committed?

**Comment by Chii-Onan** [comment]
> Looks like there is quite a lot of errors :P Is there a way somehow to connect eslint with it aswell? So we can see the errors before they are committed?

Hi, from what I understand, this action can also run on commits, but afaik not before them, I would suggest using something like pre-commit, not sure how that would be included with the repo for contribs. I don't mind researching and including it in this PR if the community wants this!

Also, the errors should be fixed in my other PR, it's just that this branch isn't up to date

**Comment by Chii-Onan** [comment]
> May be we can use this plugin https://www.npmjs.com/package/eslint-plugin-spellcheck

Hi, thank you for suggesting this plugin, I have made the required changes to support this! (definitely didn't copy their documentation...)

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = {
     'plugin:promise/recommended',
   ],
   ignorePatterns: ['.eslintrc.js', '*.json', 'jest.config.js'],
-  plugins: ['import', 'promise', '@typescript-eslint', 'prettier'],
+  plugins: ['import', 'promise', '@typescript-eslint', 'prettier', 'spellcheck'],
```
Comment (scopsy) [inline .eslintrc.js:12]: Should this plugin also be installed on globally as a dependency?

**Comment by scopsy** [comment]
@Chii-Onan I meant the `spellcheck` plugin, I think it should be installed as a package locally or am I mistaken here?

**Comment by scopsy** [comment]
Also the typo check action crashes with a lot of typos in the repo. How should we deal with it?

**Comment by Chii-Onan** [comment]
> @Chii-Onan I meant the `spellcheck` plugin, I think it should be installed as a package locally or am I mistaken here?

Hi, sorry but I don't exactly understand by installing the package locally, from what I understand you mean global deps right?

> Also the typo check action crashes with a lot of typos in the repo. How should we deal with it?

All the files causing errors right now were files that weren't supposed to be checked in the first place. In my last commit I have made changes so this hopefully doesn't happen again.

**Comment by Chii-Onan** [comment]
Hi  @scopsy , I believe that I have made the required changes to be able to download the package locally.

**Comment by Chii-Onan** [comment]
Hey @scopsy , could you please trigger  the action again for testing purposes? Also could you please review this pr?

**Comment by scopsy** [comment]
@Chii-Onan still looks like failing :( Also I think we need to run pnpm install in the root to generate a new lock file so the test API will pass here :)

**Review by scopsy — APPROVED** [review]

**Comment by scopsy** [comment]
Thank you for the contribution @Chii-Onan, i've switched to use the existing cspell library we used in the monorepo and updated some of the errors appeared. Really appreciate your contribution for this functionality, I'm sure it will benefit all of us 🎉

**Pull request merged**